lib/Unexpected.js

```diff
@@ -1207,18 +1206,24 @@ Unexpected.prototype._expect = function expect(context, args) {
 
     try {
         var result = executeExpect(context, subject, testDescriptionString, Array.prototype.slice.call(args, 2));
-        if (isPendingPromise(result)) {
-            that.expect.notifyPendingPromise(result);
-            result = result.then(undefined, function (e) {
-                if (e && e._isUnexpected && context.level === 0) {
-                    that.setErrorMessage(e);
-                }
-                throw e;
-            });
-        } else {
-            if (!result || typeof result.then !== 'function') {
-                result = makePromise.resolve(result);
+        if (utils.isPromise(result)) {
+            if (typeof result.isPending !== 'function') {
```

Comment:
Looks like this could be wrapped up in a construct like `result = unexpectifyPromiseIfNecessary(result)`?